ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -31,16 +31,38 @@
 #include <azure/core.hpp>
 #include <azure/storage/blobs.hpp>
 
+#include <stdlib.h>
+#include <iostream>
+#include <vector>
+#include <string>
+
 azure_connection::azure_connection(const std::string &bucket_name, const std::string &obj_prefix)
     : _azure_client(Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(
         std::getenv("AZURE_STORAGE_CONNECTION_STRING"), bucket_name)),
       _bucket_name(bucket_name), _object_prefix(obj_prefix)
 {
 }
 
+/*
+ * Lists all of the objects (blobs) on the bucket (container) utilising the listBlobs function form
+ * the Azure SDK
+ */
 int
-azure_connection::list_objects(std::vector<std::string> &objects) const
+azure_connection::list_objects(
+  const std::string &prefix, std::vector<std::string> &objects, bool list_single) const
 {
+    objects::clear();
+    
+    Azure::Storage::Blobs::ListBlobsOptions blob_parameters;
+    blob_parameters.Prefix = prefix;
+
+    auto list_blobs_response = _azure_client.ListBlobs(blob_parameters);
+
+    for (auto blob_item : list_blobs_response.Blobs) {
+        objects.push_back(blob_item.Name);
+        if (list_single)
```

Comment:
Could we pass in a parameter to list the number of objects that we can list? If so, we can remove this if statement.

Reply:
I've replaced the if condition and accounted for the list_single in the blob_parameters: 
```
   if (list_single)
    blob_parameters.PageSizeHint = 1;
```
Where [pagesizehint](https://azuresdkdocs.blob.core.windows.net/$web/cpp/azure-storage-blobs/12.0.0/struct_azure_1_1_storage_1_1_blobs_1_1_list_blobs_options.html) indicates the maximum number of blobs to return